### PR TITLE
[SYCL][NATIVECPU] Support reqd_work_group_size on Native CPU

### PR DESCRIPTION
**Pull request by PietroGhg**
Adds support for the `reqd_work_group_size` attribute on Native CPU. The change in the driver is required in order to have the metadata node available in the runtime.
UR PR: https://github.com/oneapi-src/unified-runtime/pull/1477

### CODE REVIEW

**Review by mdtoguchi — APPROVED** [review]
LGTM

**Review by uwedolinsky — APPROVED** [review]
Looks good to me - I've left comments on the corresponding UR PR.

File: sycl/plugins/unified_runtime/CMakeLists.txt [hunk 1]
```diff
@@ -124,8 +124,8 @@ if(SYCL_PI_UR_USE_FETCH_CONTENT)
   )
 
   fetch_adapter_source(native_cpu
-    ${UNIFIED_RUNTIME_REPO}
-    ${UNIFIED_RUNTIME_TAG}
+    "https://github.com/PietroGhg/unified-runtime.git"
+    pietro/reqd_work_group_size
```
Comment (kbenzie) [inline sycl/plugins/unified_runtime/CMakeLists.txt:128]:
```suggestion
    ${UNIFIED_RUNTIME_REPO}
    # commit 50452b75f0d1d7434f5beb4febd642af59a994ff
    # Merge: 78b8e3e3 fa084d0a
    # Author: Kenneth Benzie (Benie) <k.benzie@codeplay.com>
    # Date:   Wed May 15 10:37:42 2024 +0100
    #     Merge pull request #1477 from PietroGhg/pietro/reqd_work_group_size
    #     [NATIVECPU] Support reqd_work_group_size on Native CPU
    50452b75f0d1d7434f5beb4febd642af59a994ff
```

**Review by kbenzie — COMMENTED** [review]
https://github.com/oneapi-src/unified-runtime/pull/1477 has now been merged, please update the native cpu tag as suggested.

File: sycl/plugins/unified_runtime/CMakeLists.txt (hunk 1 above)
Comment (PietroGhg) [reply sycl/plugins/unified_runtime/CMakeLists.txt]: Thanks @kbenzie, that's done

**Review by kbenzie — APPROVED** [review]

**Pull request merged**